src/rules/page/content/resource-not-found.js

```diff
@@ -0,0 +1,24 @@
+import {RULE_LEVELS} from '../../const'
+import {
+  checkResourceUrlExists,
+  checkExternalUrlExists,
+  collectAllLinks
+} from '../../../helpers'
+
+export default {
+  // {section, content, assessmentIds, assessmentsById}
+  action: async ({content}) => {
+    const promises = collectAllLinks(content).filter(linkInfo => !!linkInfo.url)
```

Comment:
I have seen that `collectAllLinks`  could return `null` items - it is possible place for `Cannot read properties of null (reading 'url')`

Reply:
not null, but 

`{url: null}`